build.go

```diff
@@ -284,13 +285,26 @@ func Build(f BuildFunc, options ...Option) {
 		}
 
 		var launch struct {
-			Processes []Process  `toml:"processes"`
-			Slices    []Slice    `toml:"slices"`
-			Labels    []label    `toml:"labels"`
-			BOM       []BOMEntry `toml:"bom"`
+			Processes       []Process       `toml:"processes,omitempty"`
+			DirectProcesses []DirectProcess `toml:"processes,omitempty"`
+			Slices          []Slice         `toml:"slices"`
+			Labels          []label         `toml:"labels"`
+			BOM             []BOMEntry      `toml:"bom"`
+		}
+
+		if apiVersion.LessThan(apiV09) {
+			launch.Processes = result.Launch.Processes
+		} else {
+			launch.DirectProcesses = []DirectProcess{}
+			if result.Launch.Processes != nil {
+				for _, process := range result.Launch.Processes {
+					launch.DirectProcesses = append(launch.DirectProcesses, process.ToDirectProcess())
```

Comment:
My understanding is that API v0.9 pushes the decision to use a shell back to the individual builpack instead of deciding it centrally in the `lifecycle` (via indirect process). If a buildpack maintainer consciously decides to update to v0.9 maybe it's better to consciously decide on what and how processes should be started, instead of again deciding it centrally in `packit`. On the other hand side, `packit` is more high-level library from which uses might expect that level of convenience of auto-conversion.

This conversion could also be added in a separate PR on demand.

Reply:
Thanks both for weighing in, it is an interesting problem :). I think Rob's point that we have the previous experience of failing fast in the `v0.5` API default process example he provided is good enough reason to move forward without auto process conversion.
I totally see the value in doing the conversion, but it's probably more important to keep the packit experience consistent.

Reply:
No rush!